pom.xml

```diff
@@ -4,32 +4,171 @@
     <licenses>
         <license>
             <name>MIT License</name>
             <url>http://opensource.org/licenses/MIT</url>
         </license>
     </licenses>
-    <!-- Assuming you want to host on @jenkinsci:
-    <scm>
-        <connection>scm:git:git://github.com/jenkinsci/${project.artifactId}-plugin.git</connection>
-        <developerConnection>scm:git:git@github.com:jenkinsci/${project.artifactId}-plugin.git</developerConnection>
-        <url>https://github.com/jenkinsci/${project.artifactId}-plugin</url>
-    </scm>
-    -->
+    <!-- jenkins project global properties -->
+    <properties>
+        <jenkins.version>2.60.3</jenkins.version>
+        <java.level>8</java.level>
+        <log4j.version>2.11.1</log4j.version>
+        <log4j-audit.version>1.0.0</log4j-audit.version>
+    </properties>
+
+    <!-- defines all project dependencies -->
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-audit-api</artifactId>
+            <version>${log4j-audit.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
+            <version>3.8.1</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-api</artifactId>
+            <version>${log4j.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-core</artifactId>
+            <version>${log4j.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-catalog-api</artifactId>
+            <version>${log4j-audit.version}</version>
+        </dependency>
+    </dependencies>
+
+    <!-- handle the maven build lifecycle phases -->
+    <build>
+        <plugins>
+            <!-- handles creation of jenkins plugin packages -->
+            <plugin>
+                <groupId>org.jenkins-ci.tools</groupId>
+                <artifactId>maven-hpi-plugin</artifactId>
+            </plugin>
+            <!-- log4j audit maven plugin -->
+            <plugin>
+                <groupId>org.apache.logging.log4j</groupId>
+                <artifactId>log4j-audit-maven-plugin</artifactId>
+                <version>${log4j-audit.version}</version>
+                <executions>
+                    <execution>
+                        <id>generate</id>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>generate</goal>
+                        </goals>
+                        <configuration>
+                            <packageName>io.jenkins.plugins.audit.event</packageName>
+                            <enterpriseId>18060</enterpriseId>
```

Comment:
Please don't use the ASF's enterprise id here. Back in gitter, I had previously suggested using `38143` as it's the CloudBees id which is more appropriate since Jenkins doesn't have its own.

Reply:
alright, will switch it to CloudBee's

Reply:
making the refactors for it and the maven-enforcer-plugin in the next commit.